via/services/youtube_transcript.py

```diff
@@ -145,4 +147,8 @@ def strip_html(xml_string):
 
 
 def factory(_context, request):
-    return YouTubeTranscriptService(http_service=request.find_service(HTTPService))
+    session = requests.Session()
+    if youtube_proxy := request.registry.settings.get("youtube_proxy"):
+        session.proxies["https"] = youtube_proxy
+    http_svc = HTTPService(session=session)
+    return YouTubeTranscriptService(http_service=http_svc)
```

Comment:
We'll use the proxy for all requests of the service, we might have to  change that in the future but this does the tricky for now :+1:

Reply:
This is a good point. There are two requests this service makes:

- The https://youtubei.googleapis.com/youtubei/v1/player endpoint which returns JSON containing signed URLs for caption tracks. This is the one where the proxy is needed.
- Fetching the caption track URLs which are XML content containing the captions. I don't know if the proxy is essential for this.